tests/integration-tests/tests/common/utils.py

```diff
@@ -94,6 +97,29 @@ def retrieve_latest_ami(region, os, ami_type="official", architecture="x86_64"):
         raise
 
 
+def retrieve_pcluster_ami_with_additional_filters(os, architecture, additional_filters):
+    try:
+        ami_name = "aws-parallelcluster-{version}-{ami_name}".format(
+            version=get_installed_parallelcluster_version(),
+            ami_name=OS_TO_PCLUSTER_AMI_NAME_OWNER_MAP.get(os).get("name"),
+        )
+        response = boto3.client("ec2").describe_images(
+            Filters=[{"Name": "name", "Values": [ami_name]}, {"Name": "architecture", "Values": [architecture]}]
+            + additional_filters,
+            Owners=PCLUSTER_AMI_OWNERS,
+        )
+        return response["Images"][0]["ImageId"]
+    except ClientError as e:
+        LOGGER.critical(e.response.get("Error").get("Message"))
+        raise
+    except AttributeError as e:
+        LOGGER.critical("Error no attribute {0} in dict: {1}".format(os, e))
+        raise
+    except IndexError as e:
+        LOGGER.critical("Error no ami retrieved: {0}".format(e))
```

Comment:
I think in pcluster3 we use fstrings. We can use fstrings everywhere rather than format. You can see here [why.](https://realpython.com/python-f-strings/)

Reply:
additional best practice: when using a logging object it is better to use lazy logging evaluation so that no concatenation or string formatting is performed if the log level is lower than the configured one. This means that rather than using format or f-strings with the logger object you want to default to the logging specific formatting capabilities.

https://pycodequ.al/docs/pylint-messages/w1201-logging-not-lazy.html